src/components/PopupWithForm.js

```diff
@@ -22,9 +23,31 @@ export default class PopupWithForm extends Popup {
     this._modalForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
       const inputValues = this._getInputValues();
-      console.log("Input values before submit:", inputValues);
+      this.renderSaving(true);
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.